### PR TITLE
Displaying Application template name in the applications list of console

### PR DESCRIPTION
**Pull request by malshan1998**
### Related Issues
- Issue `#1` https://github.com/wso2/product-is/issues/10681#issue-757081311
- Issue `#2` https://github.com/wso2/product-is/issues/13944
- Issue `#3` https://github.com/wso2/product-is/issues/14114

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/application-list.tsx [hunk 1]
```diff
@@ -339,6 +333,68 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
+                key: "templateID",
+                render: (app: ApplicationListItemInterface): ReactNode => {
+
+                    // Note: the templateId for Standard-Based Applications in applicationTemplates is 'custom-application' (only 1 template is available)
+                    // But backend passes 3 distinct ids for Standard Based Applications
+                    // So, a deep clone of a template with templateId === 'custom-application' should be made
+                    // And change the name according to the templateId passed from the backend
+
+                    // Create a set with custom-application Ids
+                    const customApplicationIds = new Set( [
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/application-list.tsx:349]:
```suggestion
                    const customApplicationIds = new Set([
```

File: apps/console/src/features/applications/pages/applications.tsx [hunk 2]
```diff
@@ -144,7 +144,7 @@ const ApplicationsPage: FunctionComponent<ApplicationsPageInterface> = (
         isLoading: isApplicationListFetchRequestLoading,
         error: applicationListFetchRequestError,
         mutate: mutateApplicationListFetchRequest
-    } = useApplicationList(listItemLimit, listOffset, searchQuery);
+    } = useApplicationList("advancedConfigurations,templateId",listItemLimit, listOffset, searchQuery);
```
Comment (yasinmiran) [inline apps/console/src/features/applications/pages/applications.tsx:147]:
```suggestion
    } = useApplicationList("advancedConfigurations,templateId", listItemLimit, listOffset, searchQuery);
```

File: apps/console/src/features/applications/components/application-list.tsx [hunk 3]
```diff
@@ -339,6 +333,68 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
+                    } else {
+                        // Displaying template name if it's not a Standard-Based Application
+                        const template = applicationTemplates
+                            && applicationTemplates instanceof Array
+                            && applicationTemplates.length > 0
+                            && applicationTemplates.find((template) => template.id === app.templateId);
+
+                        return (
+                            <div>
+                                <p>{template?.name}</p>
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/application-list.tsx:391]: A `div` would suffice here. No need a `p` wrap.

**Review by yasinmiran — CHANGES_REQUESTED** [review]

File: apps/console/src/features/applications/components/application-list.tsx [hunk 4]
```diff
@@ -255,7 +258,9 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
      *
      * @return {TableColumnInterface[]}
      */
+
```
[the remark this answers is not in the document]
Comment (malshan1998) [reply apps/console/src/features/applications/components/application-list.tsx:261]: Done

File: apps/console/src/features/applications/components/application-list.tsx [hunk 5]
```diff
@@ -255,7 +258,9 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
     const resolveTableColumns = (): TableColumnInterface[] => {
+        console.log("resolveTableColumns called")
```
[the remark this answers is not in the document]
Comment (malshan1998) [reply apps/console/src/features/applications/components/application-list.tsx:263]: Done

File: apps/console/src/features/applications/components/application-list.tsx [hunk 6]
```diff
@@ -38,6 +38,8 @@ import {
     TableColumnInterface,
     useConfirmationModalAlert
 } from "@wso2is/react-components";
+// eslint-disable-next-line no-restricted-imports
+import _ from "lodash";
```
[the remark this answers is not in the document]
Comment (malshan1998) [reply apps/console/src/features/applications/components/application-list.tsx:42]: Done

File: apps/console/src/features/applications/components/application-list.tsx (hunk 1 above)
Comment (malshan1998) [reply apps/console/src/features/applications/components/application-list.tsx:349]: Done

File: apps/console/src/features/applications/components/application-list.tsx (hunk 3 above)
Comment (malshan1998) [reply apps/console/src/features/applications/components/application-list.tsx:391]: Done

File: apps/console/src/features/applications/pages/applications.tsx (hunk 2 above)
Comment (malshan1998) [reply apps/console/src/features/applications/pages/applications.tsx:147]: Done

File: apps/console/src/features/applications/components/application-list.tsx [hunk 7]
```diff
@@ -339,6 +333,68 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
+                        const template = applicationTemplates
+                            && applicationTemplates instanceof Array
+                            && applicationTemplates.length > 0
+                            && applicationTemplates.find((template) => {
+                                return template.templateId === "custom-application";
+                            });
+
+                        // Cloning the template and replacing the name with specific template name
+                        const template_clone = _.cloneDeep(template);
+                        if (template_clone) {
```
Comment (malshan1998) [inline apps/console/src/features/applications/components/application-list.tsx:367]: templateClone

File: apps/console/src/features/applications/components/application-list.tsx [hunk 8]
```diff
@@ -339,6 +333,68 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
+                    if (customApplicationIds.has(app.templateId)) {
+                        const template = applicationTemplates
+                            && applicationTemplates instanceof Array
+                            && applicationTemplates.length > 0
+                            && applicationTemplates.find((template) => {
+                                return template.templateId === "custom-application";
+                            });
+
+                        // Cloning the template and replacing the name with specific template name
+                        const template_clone = _.cloneDeep(template);
```
Comment (malshan1998) [inline apps/console/src/features/applications/components/application-list.tsx:366]: interface (applicationTemplates)

File: apps/console/src/features/applications/components/application-list.tsx [hunk 9]
```diff
@@ -339,6 +333,68 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
...
+                    ]);
+
+                    // Checking whether the templateId from backend, is for a custom application
+                    // If so, find the single template from applicationTemplates
+                    if (customApplicationIds.has(app.templateId)) {
+                        const template = applicationTemplates
+                            && applicationTemplates instanceof Array
+                            && applicationTemplates.length > 0
+                            && applicationTemplates.find((template) => {
+                                return template.templateId === "custom-application";
```
Comment (malshan1998) [inline apps/console/src/features/applications/components/application-list.tsx:362]: "custom-application" : json

**Comment by malshan1998** [comment]
Closing due to errors in version control. Will resolve the errors in a separate branch and send a PR.

**Pull request closed without merging**